app/src/main/java/com/hedvig/app/feature/embark/passages/externalinsurer/ExternalInsurerFragment.kt

```diff
@@ -3,30 +3,91 @@ package com.hedvig.app.feature.embark.passages.externalinsurer
 import android.os.Build
 import android.os.Bundle
 import android.view.View
+import androidx.activity.result.ActivityResult
+import androidx.activity.result.contract.ActivityResultContracts
 import androidx.core.view.doOnNextLayout
+import androidx.core.view.isVisible
 import androidx.fragment.app.Fragment
+import androidx.fragment.app.setFragmentResultListener
+import androidx.lifecycle.flowWithLifecycle
+import androidx.lifecycle.lifecycleScope
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
 import com.hedvig.app.R
 import com.hedvig.app.databinding.PreviousOrExternalInsurerFragmentBinding
+import com.hedvig.app.feature.embark.EmbarkViewModel
 import com.hedvig.app.feature.embark.passages.MessageAdapter
 import com.hedvig.app.feature.embark.passages.externalinsurer.askforprice.AskForPriceInfoActivity
+import com.hedvig.app.feature.embark.passages.externalinsurer.askforprice.AskForPriceInfoActivity.Companion.RESULT_SKIP
 import com.hedvig.app.feature.embark.passages.externalinsurer.askforprice.AskForPriceInfoParameter
-import com.hedvig.app.feature.embark.passages.previousinsurer.PreviousInsurerBottomSheet
+import com.hedvig.app.feature.embark.passages.previousinsurer.InsurerProviderBottomSheet
+import com.hedvig.app.feature.embark.passages.previousinsurer.PreviousInsurerParameter
+import com.hedvig.app.util.extensions.showErrorDialog
 import com.hedvig.app.util.extensions.view.setupInsetsForIme
 import com.hedvig.app.util.whenApiVersion
 import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
+import kotlinx.coroutines.flow.launchIn
+import kotlinx.coroutines.flow.onEach
 import org.koin.androidx.viewmodel.ext.android.sharedViewModel
 
 class ExternalInsurerFragment : Fragment(R.layout.previous_or_external_insurer_fragment) {
 
     private val binding by viewBinding(PreviousOrExternalInsurerFragmentBinding::bind)
+
+    private val embarkViewModel: EmbarkViewModel by sharedViewModel()
     private val viewModel: ExternalInsurerViewModel by sharedViewModel()
 
+    private val askForPriceActivityResultLauncher =
+        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) { result: ActivityResult ->
+            if (result.resultCode == RESULT_SKIP) {
+                onContinue()
```

Comment:
I'm a bit unsure how we handle "switching" here, that was previously done with `PreviousInsurer`. @sampettersson  how is this implemented on iOS? Eg. A users selects "skip" in insurely, but has still selected their previous insurer. Are we showing switching information in the offer screen?

Reply:
Yes, you should still set the previous insurer in the store (don't set the dataCollectionId though), regard skipping as just saying "I don't want to use insurely, but I still want to switch".